### PR TITLE
TST: Fix excel test discovery

### LINKED ISSUE #16477: Some excel tests aren't being picked up
Discovered by @abarber4gh 

```
pytest pandas/tests/io/test_excel.py -v
=================================================================== test session starts ====================================================================
platform darwin -- Python 3.6.1, pytest-3.1.0, py-1.4.33, pluggy-0.4.0 -- /Users/taugspurger/Envs/pandas-dev/bin/python3.6
cachedir: .cache
rootdir: /Users/taugspurger/Envs/pandas-dev/lib/python3.6/site-packages/pandas, inifile: setup.cfg
plugins: xdist-1.15.0, cov-2.4.0
collected 5 items

pandas/tests/io/test_excel.py::test_styler_to_excel[xlwt] xfail
pandas/tests/io/test_excel.py::test_styler_to_excel[xlsxwriter] PASSED
pandas/tests/io/test_excel.py::test_styler_to_excel[openpyxl] PASSED
pandas/tests/io/test_excel.py::TestFSPath::test_excelfile_fspath PASSED
pandas/tests/io/test_excel.py::TestFSPath::test_excelwriter_fspath PASSED

=========================================================== 4 passed, 1 xfailed in 0.97 seconds ============================================================
```

probably related to the pytest move. PR coming shortly

### PR DESCRIPTION
**Pull request by TomAugspurger**
closes #16477 

Locally:

```
======================================== 26 failed, 358 passed, 48 skipped, 1 xfailed, 61 warnings in 20.85 seconds ========================================
```

### CODE REVIEW

**Comment by jreback** [comment]
oh, so these were not running? hope they pass!

File: pandas/io/excel.py [hunk 1]
```diff
@@ -202,6 +202,16 @@ def read_excel(io, sheet_name=0, header=0, skiprows=None, skip_footer=0,
                dtype=None, true_values=None, false_values=None, engine=None,
                squeeze=False, **kwds):
 
+    # Can't use _deprecate_kwarg since sheetname=None has a special meaning
+    if sheet_name is 0 and 'sheetname' in kwds:
```
Comment (dsm054) [inline pandas/io/excel.py]: `is` is for identity. ;-)

Comment (TomAugspurger) [inline pandas/io/excel.py]: I think we want that here. `sheet_name` could be a str, int, or None and we don't want to raise a TypeError. Just checking whether they changed the default.

Comment (TomAugspurger) [inline pandas/io/excel.py]: Although, is that a CPython specific thing, pre-defining some integer objects? Maybe I should change it...

Comment (TomAugspurger) [inline pandas/io/excel.py]: Looks like it is specific to CPython. I'll update.

Comment (dsm054) [inline pandas/io/excel.py]: Yeah, `is` only works at all because of the int cache.  It's almost never what anyone wants.

**Comment by jreback** [comment]
ahh this is fixing up after #16442?

**Comment by TomAugspurger** [comment]
Yeah, 11bed216c3021e19618a9c65e6c0430cdf887826 fixes that. Would have been an unfortunate breakage :/

**Comment by TomAugspurger** [comment]
Ugh. That random matplotlib failure is spreading: https://ci.appveyor.com/project/pandas-dev/pandas/build/1.0.2225/job/6lg79jfhas0l9qiw#L1128 (I hadn't seen it on windows yet)

@jreback ok to merge?

**Comment by jreback** [comment]
yep

**Comment by jorisvandenbossche** [comment]
@TomAugspurger You will have to split this commit if you want to backport this PR, as the sheetname change is only targetted for 0.21.0

**Comment by TomAugspurger** [comment]
> @TomAugspurger You will have to split this commit if you want to backport this PR, as the sheetname change is only targetted for 0.21.0

I hadn't thought of that... I think I'll not backport it then, unless there are objections.

**Comment by jorisvandenbossche** [comment]
It's only about some excel test not running on the 0.20.x branch? That sounds OK to me to not backport if we don't have any excel related changes in 0.20.x

**Pull request merged**